sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1229,3 +1229,132 @@ case class Flatten(child: Expression) extends UnaryExpression {
+      null
+    } else {
+      new GenericArrayData(List.fill(count.asInstanceOf[Int])(left.eval(input)))
+    }
+  }
+
+  override def prettyName: String = "array_repeat"
+
+  override def nullSafeCodeGen(ctx: CodegenContext,
+                               ev: ExprCode,
+                               f: (String, String) => String): ExprCode = {
+    val leftGen = left.genCode(ctx)
+    val rightGen = right.genCode(ctx)
+    val resultCode = f(leftGen.value, rightGen.value)
+
+    if (nullable) {
+      val nullSafeEval =
+        leftGen.code +
+          rightGen.code + ctx.nullSafeExec(right.nullable, rightGen.isNull) {
+            s"""
+              ${ev.isNull} = false;
+              $resultCode
+            """
+          }
+
+      ev.copy(code = s"""
+        boolean ${ev.isNull} = true;
+        ${CodeGenerator.javaType(dataType)} ${ev.value} = ${CodeGenerator.defaultValue(dataType)};
+        $nullSafeEval
+      """)
+    } else {
+      ev.copy(code = s"""
+        boolean ${ev.isNull} = false;
+        ${leftGen.code}
+        ${rightGen.code}
+        ${CodeGenerator.javaType(dataType)} ${ev.value} = ${CodeGenerator.defaultValue(dataType)};
+        $resultCode""", isNull = FalseLiteral)
+    }
+
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+
+    nullSafeCodeGen(ctx, ev, (l, r) => {
+      val et = dataType.elementType
+      val isPrimitive = CodeGenerator.isPrimitiveType(et)
+
+      val arrayDataName = ctx.freshName("arrayData")
+      val arrayName = ctx.freshName("arrayObject")
+      val numElements = ctx.freshName("numElements")
+
+      val genNumElements =
+        s"""
+           | int $numElements = 0;
+           | if ($r > 0) {
+           |   $numElements = $r;
+           | }
+         """.stripMargin
+
+      val initialization = if (isPrimitive) {
+        val arrayName = ctx.freshName("array")
+        val baseOffset = Platform.BYTE_ARRAY_OFFSET
+        s"""
+           | int numBytes = ${et.defaultSize} * $numElements;
+           | int unsafeArraySizeInBytes = UnsafeArrayData.calculateHeaderPortionInBytes($numElements)
+           |   + org.apache.spark.unsafe.array.ByteArrayMethods
+           |     .roundNumberOfBytesToNearestWord(numBytes);
+           | byte[] $arrayName = new byte[unsafeArraySizeInBytes];
+           | UnsafeArrayData $arrayDataName = new UnsafeArrayData();
+           | Platform.putLong($arrayName, $baseOffset, $numElements);
+           | $arrayDataName.pointTo($arrayName, $baseOffset, unsafeArraySizeInBytes);
+           | ${ev.value} = $arrayDataName;
+         """.stripMargin
+      } else {
+        s"${ev.value} = new ${classOf[GenericArrayData].getName()}(new Object[$numElements]);"
+      }
+
+      val primitiveValueTypeName = CodeGenerator.primitiveTypeName(et)
+      val assignments = {
+        val updateArray = if (isPrimitive) {
+          val isNull = left.genCode(ctx).isNull
+          s"""
+             | if ($isNull) {
```

Comment:
I thought it could be, in this case?
```
    val intDF = {
      val schema = StructType(Seq(
        StructField("a", IntegerType),
        StructField("b", IntegerType)))
      val data = Seq(
        Row(1, 1),
        Row(3, 2),
        Row(null, 2)
      )
      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
    }
```

Reply:
Ah, sorry for my misunderstanding. `isPrimitive` does not stand for `element is non-null`.